### DISCUSSION TITLE
Double menù

### DISCUSSION THREAD

**Discussion by itasterix** [templaza/astroid-framework#336] [Q&A]
Hello
I need to use Astroid on a web site that actually has an upper menù, a logo and the main menù.
I was not able to replicate using the template, 
The upper menù does non work in the header section and also if I try to place in the Contact Information section. 
It always appears vertical but I want it horizontal.

Many thanks

![immagine](https://github.com/templaza/astroid-framework/assets/108941650/f486be2d-d6bb-4bf4-958a-4fc99b0c5556)

**Comment by Chacapamac** [comment]
I @itasterix - I will try to help You!
Fiirst you have to understand that only one menu will appear in the Sticky Header, the one you set in the Astroid Templete Manager “Header” section.

**You mean something like that?** —> http://joom4-astro.graf-test.ca/

**Reply by itasterix** [reply]
Yes, thanks! The site you linked is a good example

**Comment by Chacapamac** [comment]
I @itasterix !
 The website you see is only a test website.

**This is what I did**

**1.** I create a new “**Section**” _(position)_ to have a better control and move it to the top of the normal top “**Header**” position

![New Section](https://github.com/templaza/astroid-framework/assets/1907276/e843ad03-e68b-48bc-8dc5-99c2905e0afe)

![move-on-top](https://github.com/templaza/astroid-framework/assets/1907276/46136fe4-58a0-4f65-b26a-4456c233576b)


**2.** Create your menu and assign to a new menu module assign to a new custom layout position

![create-module-position](https://github.com/templaza/astroid-framework/assets/1907276/6bc26e30-89b6-4a31-a477-edc5aafae5a5)


**3.** Return to your New top custom position in the Layout Manager and add your Menu Module to the position

![add-module-position](https://github.com/templaza/astroid-framework/assets/1907276/f06ab9df-8d79-4b58-8dd3-f1d69b0c2f25)


**4.** Change the “**Header**” Settings

![Header-settings](https://github.com/templaza/astroid-framework/assets/1907276/74f6cab4-07a8-4e3d-a75e-4dc28bdc423a)


Remember that if you need that menu to appear in the mobile menu (merging both menu) you will have to add to the ”**Main Menu**” your identical created top menu to it’s menu items and hide these menu items on other devices than mobile. 

**Just add the custom Bootstrap classes** “.d-block .d-md-none”  to all elements of your top menu. 

You can even at this point make all that custom top menu a submenu with an easy title like “**My Top Menu**” as a top element in the “**Main Menu**” to separate the 2 merged menus on mobile (if Neede}

If Needed add margin, padding, background color, special color for your links, text in the Top Menu custom position in the layout manager

If your menu is not horizontal go on your menu module and assign the “HorizontalView” Layout from your own template

![horixontal](https://github.com/templaza/astroid-framework/assets/1907276/77760d79-1436-4589-8fc3-f9e748a938ab)

**Comment by itasterix** [comment]
Thank you so much for your kindly answer!!

I try to follow your precious guide but I still have some issue.

I have added a new section:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/b47ed5d1-1c1b-47d0-9329-960ba4fa6ef6)

and added the class to all menu items:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/f9de7083-2d4c-41fd-a73f-5b8a3906ea57)

also try to add the class here, **but there is not the option for the HORIZONTAL LAYOUT you told**:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/4ab058f2-2a28-47d8-ba6d-9222688298b4)

Still the result I have is not an horizontal menù:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/fe74d3d3-5246-4829-84e2-7732c06c4b29)

Can you help me to understand why?

**Reply by Chacapamac** [reply]
@itasterix 
**Ok, I see, you want to achieve that**
![Screen Shot 2023-05-24 at 8 54 03 AM](https://github.com/templaza/astroid-framework/assets/1907276/610e985e-de3b-4844-9e75-687e1ddd9357)

In fact, the result you show me just above **is an horizontal menu with vertical sub-menus** , just like the Main Astroid Menu but missing the css and formatting of the Main menu.

![Screen Shot 2023-05-24 at 8 59 20 AM](https://github.com/templaza/astroid-framework/assets/1907276/f9b22938-efa8-48e9-a5c1-aeb3f8774a6c)

**I try and I’m not sure how to repeat the formatting of the main Astroid menu on your custom menu, easily.**
**I will ask myself the question to the devs, maybe they have an easy solution.**

**JUST A NOTE**:  on adding class to module in Joomla and the Bootstrap hiding classes needed if you want to incorporate this custom menu to the mobile menu.
• If you need to incorporate your custom menu to the mobile menu, YOU NEED to RE-CREATE this menu INSIDE the MAIN Astroid MENU and hide all the custom menu items on larger device than phone with adding the hiding classes [SPACE]d-block d-md-none  (No dots)

![Screen Shot 2023-05-24 at 8 31 03 AM](https://github.com/templaza/astroid-framework/assets/1907276/c028c2fd-3367-4a2d-a681-1ad97c1fc3d1)

**Reply by itasterix** [reply]
Thank you so much for your support! I really appreciate
I 'll wait for an answer if horizontal menu with vertical sub-menus is possible to achieve.

**Comment by Chacapamac** [comment]
@itasterix 
**horizontal menu with vertical sub-menus** is what you already have in your custom Top Menu. (as shown on my last post)
Simply, the correct formatting that apply to the “Main Astroid Menu is missing.

I already ask the question how to add the formatting. I will also ask if its possible to add the ability to add a second menu in the Header to Astroid Layout manager. (That will simplify all that)

**Reply by itasterix** [reply]
Thank you very much, I'll wait !

**Comment by itasterix** [comment]
Hello, I decided to move the "guide" section to main an reorganize it.
So I will have a simple and flat "Top Menù" , waitin g for a simple solution.
The issue now is that single voices of Top menù are not spacing between them.

Why it happens? What I have missing?

![immagine](https://github.com/templaza/astroid-framework/assets/108941650/42d18a8a-372b-4dd6-8e58-e29d3bd29bf7)

**Comment by Chacapamac** [comment]
@itasterix 
Yeah, I always try to avoid 2 menus, always a bit confusing for users. 

I will ask for the ability to have an easy way in Astroid to have a second menu formatted the exact same way as the “Main” menu without having to use coding 
**IDEAS:** Set in Astroid Template Manager?, Just by adding a class?, having a special Astroid Menu module?...) 

**For your top menu squish together** it’s because you do not have any CSS formatting on the **ul > li** forming it.

• If you have no ability to choose a **Menu Module layout** like the “**HorizontalView**” from **Astroid Template One or Zero or Your Astroid Template** shown here, just add a custom class e.g. “**my-top-menu**” to the **Menu Class** and make your own formatting.
e.g.
`ul.my-top-menu li { margin-right: 1rem; }`

![Screen Shot 2023-05-31 at 12 04 26 PM](https://github.com/templaza/astroid-framework/assets/1907276/acec3215-a16a-4a9a-b968-0e53d3ef1ca8)

**Comment by sonvnn** [comment]
@itasterix @Chacapamac I created a module for Astroid Menu in https://github.com/templaza/astroid-framework/commit/928c08c4886417394d25e13e23a4d0e31aadd81b . Please help me download developer branch and install on your website for testing.

You can download developer branch here https://github.com/templaza/astroid-framework/archive/refs/heads/developer.zip

Thanks & Best Regards,
Sonny

**Reply by Chacapamac** [reply]
Just test here —> http://joom4-astro.graf-test.ca/  (Only Home page)
Seems to work great and take the “Main Menu” formatting

• Submenus are not showing because clicking or hover on the first level li doesn't change classes

![Screen Shot 2023-06-02 at 9 48 47 AM](https://github.com/templaza/astroid-framework/assets/1907276/a4c7d0c6-9e78-48b7-a545-9ce793b50d1e)

• Do you think it wil be an easy way to add (or not) that menu to the mobile with a setting?

**Reply by sonvnn** [reply]
Thank you! I will retest it again and give you solution. Have a nice weekend my friend :)

**Reply by itasterix** [reply]
Hello, I have download developer branch and installed on my site, but I see no difference in menù.
Maybe I have missed something.

**Reply by sonvnn** [reply]
You have to go to Module Manager >> Add new module >> Astroid Menu.
And now you can configure it like Joomla Menu Module.

**Reply by itasterix** [reply]
> You have to go to Module Manager >> Add new module >> Astroid Menu. And now you can configure it like Joomla Menu Module.

Yes, it is what I have already tried, but with no success. It does not exist any Astroid Menu module.
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/f3fdd3ef-8db4-4c44-9aef-75c7496082b1)

I'm using Joomla 3 at the moment and I'm configuring Astroid for upgrade to Joomla 4, maybe this could be the reason.

**Reply by Chacapamac** [reply]
You have until August 17 before Joomla 3 arrive to the end of support

**Reply by sonvnn** [reply]
I fixed dropdown submenu issue in https://github.com/templaza/astroid-framework/commit/1ef471a8a0de23e9a4be96cc30aeff607383d0a5 Please help me download developer branch and retest again.

PS: Sorry for the late reply! I was busy with a new project and thought about Astroid 3 with React JS replacing Angular JS. It will improve back-end performance very much

**Reply by Chacapamac** [reply]
@sonvnn 
Just test **Astroid Menu Module** dropdown seem to work perfectly.

**QUESTION:**
Do You think it’s a possibility to be able to have a setting in the **Astroid Menu Module** to be added (or not) to the Main Mobile Menu. That will be a powerful option that will be easier than adding manually an only mobile visible copy of it to the Main Menu.

**Reply by sonvnn** [reply]
I'm sorry. It's impossible, at the moment. I will research about it and response you later.

**Comment by itasterix** [comment]
Just to let you know, I understand nothing about CSS.
I have finally found a way to let it work!

This is the correct CSS istruction:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/a7011637-29d1-42ca-bf4b-ff2963aa43bb)

And now top menù is at correct distance:
![immagine](https://github.com/templaza/astroid-framework/assets/108941650/2307d902-dd78-4c6a-9175-647e4f28d3d3)

For now it is enough for me, waiting to have a final solution.
Thank you for your support!

**Reply by Chacapamac** [reply]
In Joomla 4 with the installation of the Developer branch (link above)
![Screen Shot 2023-06-11 at 9 26 50 AM](https://github.com/templaza/astroid-framework/assets/1907276/3753d5a7-3729-481e-bad6-12dba0168cce)

The menu in that module have the exact same look as the Main Astroid menu (Horizontal, Dropdown icon...) 
— **For now the dropdown items are not showing and it is not automatically integrated in the Mobile menu**

**Reply by itasterix** [reply]
Many tanks, but I'm still on Joomla 3.
After migrating to 4 I'll give a try.

**Reply by Chacapamac** [reply]
Cool!
@sonvnn 
Whenever you want I’m ready to test the Astroid Menu Module.
Good Day to All!
